webplugin/plugin.js

```diff
@@ -316,25 +334,43 @@ function injectJquery() {
         addableDocument = iframeDocument;
         addableDocument.body.setAttribute('dir', languageDirectionChangeAuto());
     }
+
+    addableWindow.MCK_THIRD_PARTY_INTEGRATION = MCK_THIRD_PARTY_INTEGRATION;
+    addableWindow.MCK_ENV_DETAILS = MCK_ENV_DETAILS;
+
     var head = addableDocument.getElementsByTagName('head')[0];
     var script = addableDocument.createElement('script');
     script.async = false;
     script.type = 'text/javascript';
     script.src = 'https://cdn.kommunicate.io/kommunicate/jquery-3.5.1.min.js';
+
+    function loadSentry() {
+        return scriptLoader({
+            _document: addableDocument,
+            url: THIRD_PARTY_SCRIPTS.sentry.js,
+            enabled: MCK_THIRD_PARTY_INTEGRATION.sentry.enabled,
+            ignoreIfError: true,
+        })
+            .then(addKommunicatePluginToIframe)
+            .catch(function (error) {
+                console.error(error);
+            });
+    }
+
     if (script.readyState) {
         // IE
         script.onreadystatechange = function () {
             if (
                 script.readyState === 'loaded' ||
                 script.readyState === 'complete'
             ) {
-                addKommunicatePluginToIframe();
+                loadSentry();
```

Comment:
flow looks confusing and difficult for anyone to understand and navigate the code while debugging.

Reply:
adding comments